api/download.py

```diff
@@ -46,15 +46,14 @@ def _append_targets(self, targets, cont_name, container, prefix, total_size, tot
                         break
                 if filtered:
                     continue
-            if optional or not f.get('optional', False):
-                filepath = os.path.join(data_path, util.path_from_hash(f['hash']))
-                if os.path.exists(filepath): # silently skip missing files
-                    if cont_name == 'analyses':
-                        targets.append((filepath, prefix + '/' + ('input' if f.get('input') else 'output') + '/' + f['name'], cont_name, str(container.get('_id')),f['size']))
-                    else:
-                        targets.append((filepath, prefix + '/' + f['name'], cont_name, str(container.get('_id')),f['size']))
-                    total_size += f['size']
-                    total_cnt += 1
+            filepath = os.path.join(data_path, util.path_from_hash(f['hash']))
+            if os.path.exists(filepath): # silently skip missing files
```

Comment:
This is the location of the skipped missing filepaths I mentioned. I would write something similar to 
> Expected filepath <filepath> to exist but it is missing. File will be skipped in download.